utils/format_hpaRna.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+# Description: Format RNA HPA tissue gene data so that it is usable for dataOverlay in MetAtlas
+
+# Raw data can be downloaded from https://www.proteinatlas.org/download/rna_tissue_hpa.tsv.zip
+
+import sys
+import os
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+import argparse
+import yaml
+
+def scalingfunc(x, U = 10):
+    return min(1, np.log2(x + 1)/U)
+
+def logfunc(x):
+    return np.log2(x+1)
+
+def calMax(qt):
+    # input: percentile [25, 50 ,75]
+    return qt[2] + 1.5*(qt[2]-qt[0])
+
+def apply_scale(df, U):
+    dt_tpm_lg = df['TPM'].apply(scalingfunc, U=U)
+    df_lg = pd.DataFrame({ 
+        'TPM (lg, U=%g)'%(U) : dt_tpm_lg,
+        })
+    return df_lg
+
+def getGeneIdSetFromModel(modelfile):
+    """Retrieve set of gene Ids from the modelfile
+       return None of failed to parse the model file
+    """
+    geneIdList = []
+    with open(modelfile, "r") as stream:
+        try:
+            modeldata = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print("Failed to parse model file %s"%(modelfile),  file = sys.stderr)
+            return None
+    for (key, content) in modeldata:
+        if key == "genes":
+            for idlist in content:
+                for idtuple in idlist:
+                    geneIdList.append(idtuple[1])
+    geneIdSet = set(geneIdList)
+    return geneIdSet
+
+def plot(df_lg, U, figfile):
+    fig = plt.figure(figsize=(15,8))
+    gs = fig.add_gridspec(1,2)
+
+    ax1 = fig.add_subplot(gs[0,0])
+    vf = np.vectorize(scalingfunc)
+    x = np.linspace(0,100,2000)
+    y = vf(x, U)
+    ax1.set_xlabel('Raw data')
+    ax1.set_ylabel('Scaled')
+    ax1.plot(x, y, 'r')
+    ax1.set_ylim(0, 1)
+
+    ax2 = fig.add_subplot(gs[0,1])
+    ax2.autoscale()
+    ax2.set_xlabel('Scaled')
+    ax2.set_ylabel('Percentage')
+    df_lg.hist(column = df_lg.columns[0], bins = 50, ax=ax2, density=1)
+
+    bsname = figfile
+    fig.savefig('%s.png'%(bsname))
+    fig.savefig('%s.pdf'%(bsname))
+    plt.clf()
+    plt.close()
+
+def formatHpaRna(df, U, geneIdSet, outfile):
```

Comment:
It would be nice with some more doc strings in this file in general, and for long and complicated functions like this one in particular.

Reply:
Good point @MalinAhlberg , I've added some comments to these functions in commit 6552ee4